src/psyclone/tests/psyad/main_test.py

```diff
@@ -0,0 +1,260 @@
+def test_main_invalid_filename():
+    '''Test that the the main() function raises an exception if the
+    filename does not exist.
+
+    '''
+    # FileNotFoundError does not exist in Python2
+    try:
+        FileNotFoundError
+    except NameError:
+        FileNotFoundError = IOError
+    with pytest.raises(FileNotFoundError) as info:
+        main(["does_not_exist.f90"])
+    assert ("[Errno 2] No such file or directory: 'does_not_exist.f90'"
+            in str(info.value))
+
+
+# writing to stdout
+def test_main_stdout(tmpdir, capsys):
+    '''Test that the the main() function returns its output to stdout by
+    default.
+
+    '''
+    tl_code = (
+        "program test\n"
+        "integer :: a\n"
+        "a = 0.0\n"
+        "end program test\n")
+    expected = (
+        "program test\n"
+        "  integer :: a\n\n"
+        "  a = 0.0\n\n"
+        "end program test\n")
+    filename = six.text_type(tmpdir.join("tl.f90"))
+    with open(filename, "a") as my_file:
+        my_file.write(tl_code)
+    main([filename])
+    output, error = capsys.readouterr()
+    assert error == ""
+    assert expected in output
+
+
+# using -oad
+def test_main_fileout(tmpdir, capsys):
+    '''Test that the the main() function returns its output to a file if
+    specified.
+
+    '''
+    tl_code = (
+        "program test\n"
+        "integer :: a\n"
+        "a = 0.0\n"
+        "end program test\n")
+    expected = (
+        "program test\n"
+        "  integer :: a\n\n"
+        "  a = 0.0\n\n"
+        "end program test\n")
+    filename_in = str(tmpdir.join("tl.f90"))
+    filename_out = str(tmpdir.join("ad.f90"))
+    with open(filename_in, "a") as my_file:
+        my_file.write(tl_code)
+    main([filename_in, "-oad", filename_out])
+    output, error = capsys.readouterr()
+    assert error == ""
+    assert output == ""
+    with open(filename_out, 'r') as my_file:
+        data = my_file.read()
+    assert expected in data
+
+
+# -v output
+@pytest.mark.xfail(reason="issue #1235: caplog returns an empty string in "
+                   "github actions and some flavours of Python", strict=False)
+def test_main_verbose(tmpdir, capsys, caplog):
+    '''Test that the the main() function outputs additional information if
+    the -v flag is set. Actually -v seems to have no effect here as
+    pytest takes control of the logging and we have to set it to the
+    required level manually.
+
+    '''
+    tl_code = (
+        "program test\n"
+        "integer :: a\n"
+        "a = 0.0\n"
+        "end program test\n")
+    filename_in = str(tmpdir.join("tl.f90"))
+    filename_out = str(tmpdir.join("ad.f90"))
+    with open(filename_in, "a") as my_file:
+        my_file.write(tl_code)
+    with caplog.at_level(logging.DEBUG):
+        main([filename_in, "-v", "-oad", filename_out])
+
+    output, error = capsys.readouterr()
+    assert error == ""
+    assert output == ""
+    assert "INFO     root:main.py:75 Reading file /" in caplog.text
```

Comment:
Just a note that it would be nicer if we could have s/root/psyclone.psyad/

Reply:
Updated to provide the package information instead of just using root.